apps/src/code-studio/components/progress/teacherPanel/SelectedStudentPairing.jsx

```diff
@@ -0,0 +1,50 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import i18n from '@cdo/locale';
+import Tooltip from '@cdo/apps/templates/Tooltip';
+
+// Render a string and possibly a tooltip that describes the student's
+// partners. This method should only be called when the student is in a
+// pairing group. The length of partnerNames is typically equal to
+// partnerCount but the length of partnerNames can be less than
+// partnerCount if a partner's user account and/or progress was deleted.
+const SelectedStudentPairing = ({partnerNames, partnerCount}) => {
+  const renderPartners = () => {
```

Comment:
Mostly wondering, is there's an advantage to making renderPartners a function vs. just a variable here?  (While researching this, I came across [this](https://stackoverflow.com/questions/46138145/where-should-functions-in-function-components-go) which might also be relevant).  I don't have a strong opinion here, just trying to learn what the best practices and our stylistic conventions are.

Reply:
That's an interesting question, my follow-up would be, is the performance impact significant enough that it actually matters? ([see discussion here](https://stackoverflow.com/questions/57289811/function-inside-functional-component-in-react-hooks-performance)) Any functions related to updating state will be defined inside the functional components, so it might be strange to have half the functions inside the component and half outside if we decided some need to be defined outside the component.

I just copied this function from where it existed before, I did consider not having a function in this component and just rendering the output of the if/else, which may be a better way to write this component